src/server_manager/web_app/app.spec.ts

```diff
@@ -35,41 +32,43 @@ const TOKEN_WITH_ONE_SERVER = 'one-server-token';
 // tslint:disable-next-line:no-any
 (global as any).bringToFront = () => {};
 
+// Inject app-root element into DOM once before the test suite runs
+beforeAll(async () => {
+  const loadAppRoot = new AppRoot();  // TODO: Remove once figure out how to load AppRoot properly.
+
+  const appRootEl = document.createElement('app-root');
+  appRootEl.setAttribute('id', 'appRoot');
+  document.getElementsByTagName('body')[0].appendChild(appRootEl);
```

Comment:
You can write `document.body.innerHTML = "<app-root id='appRoot'></app-root>"`, which is probably cleaner.

Reply:
I like that.